error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/ImmutableSetRules.java

```diff
@@ -211,4 +214,43 @@ ImmutableSet<T> after(T e1, T e2, T e3, T e4, T e5) {
       return ImmutableSet.of(e1, e2, e3, e4, e5);
     }
   }
+
+  /** Prefer {@link Sets#intersection(Set, Set)} ()} over more contrived alternatives. */
+  static final class SetsIntersection<T> {
```

Comment:
To be honest, it's a bit hard to decide how we should name these rules. We invoke the same method in the three rules but how we get the set that we pass differs. Till now most cases also _use_ different overloads of a method and based on that we use different words.

I imagine we'd also have something that determines a suffix based on what we _pass_ as the two parameters for `Sets#intersection`. 

Was doubting whether we want to add the `ImmutableCopy` as suffix as well 🤔. 

Pushed a suggestion, let's discuss 👀.